cmd/whitesourceExecuteScan.go

```diff
@@ -657,39 +659,43 @@ func readAssessmentsFromFile(assessmentFilePath string, utils whitesourceUtils)
 	exists, err := utils.FileExists(assessmentFilePath)
 	if err != nil {
 		log.SetErrorCategory(log.ErrorConfiguration)
-		log.Entry().Errorf("unable to check existence of assessment file at '%s'", assessmentFilePath)
+		log.Entry().WithError(err).Errorf("unable to check existence of assessment file at '%s'", assessmentFilePath)
 	}
 	assessmentFile, err := utils.Open(assessmentFilePath)
 	if exists && err != nil {
 		log.SetErrorCategory(log.ErrorConfiguration)
-		log.Entry().Errorf("unable to open assessment file at '%s'", assessmentFilePath)
+		log.Entry().WithError(err).Errorf("unable to open assessment file at '%s'", assessmentFilePath)
 	}
 	assessments := &[]format.Assessment{}
 	if exists {
 		defer assessmentFile.Close()
 		assessments, err = format.ReadAssessments(assessmentFile)
 		if err != nil {
 			log.SetErrorCategory(log.ErrorConfiguration)
-			log.Entry().Errorf("unable to parse assessment file at '%s'", assessmentFilePath)
+			log.Entry().WithError(err).Errorf("unable to parse assessment file at '%s'", assessmentFilePath)
 		}
 	}
 	return assessments
 }
 
 // checkSecurityViolations checks security violations and returns an error if the configured severity limit is crossed.
-func checkProjectSecurityViolations(config *ScanOptions, cvssSeverityLimit float64, project ws.Project, sys whitesource, assessments *[]format.Assessment, influx *whitesourceExecuteScanInflux) (int, []ws.Alert, error) {
+func checkProjectSecurityViolations(config *ScanOptions, cvssSeverityLimit float64, project ws.Project, sys whitesource, assessments *[]format.Assessment, influx *whitesourceExecuteScanInflux) (int, []ws.Alert, []ws.Alert, error) {
```

Comment:
maybe it is helpful in the comment to the function to highlight the difference between the two `[]ws.Alert` which are returned. I personally have difficulties to get the difference.

2nd one seems to be the assessed alerts, here I find a speaking variable name.

For the 1st one I am not sure. Is it all the security alerts? Likely not since they seem to get filtered again. The complete list of alerts does not seem to get returned since content of `alerts` in line 685 is overwritten in line 701 by filteredAlerts.
I would assume that this is intended. If so some additional comment might be helpful or maybe just trash `alerts` and only work with `filteredAlerts`?
Maybe I am also overlooking something ...

Reply:
Added the relevant changes / additions as comments to the code